test/jdk/java/nio/channels/DatagramChannel/Connect.java

```diff
@@ -47,9 +47,10 @@ public static void main(String[] args) throws Exception {
     }
 
     static void test() throws Exception {
-        Reactor r = new Reactor();
-        Actor a = new Actor(r.port());
-        invoke(a, r);
+        try (Reactor r = new Reactor();
+             Actor a = new Actor(r.port())) {
+            invoke(a, r);
+        }
     }
```

Comment:
> Hi Conor,
> 
> Good work.
> Chris made a good remark to me privately. The test is running in same VM mode, and no longer waits for all threads (that is Actor & Reactor) to complete if one them throws an exception.
> We want, that - because we don't want the test to wait forever and timeout with no output (and no stack trace) if that happens.
> However, this may leave one of the Actor/Reactor in a state where it will live forever in the running agent VM, and may keep its DatagramChannel open - and that's bad.
> 
> One way to avoid that would be to make `Actor` & `Reactor` `Closeable`, and have them both create their `DatagramChannel` in their constructor.
> 
> Then you could modify the `test` method in this way:
> 
> ```
> static void test() throws Exception {
>         try (Reactor r = new Reactor();
>                Actor a = new Actor(r.port());) {
>             invoke(a, r);
>         }
>     }
> ```
> 
> which would ensure that both DatagramChannels are closed when the test terminate, and would ensure that the other actor/reactor is unblocked and terminates too.

@dfuch Actor & Reactor now implement AutoCloseable and are instantiated in try-with-resource arguments.